src/main/java/ch/njol/skript/expressions/ExprTime.java

```diff
@@ -32,42 +10,52 @@
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.registrations.Classes;
 import ch.njol.skript.util.Getter;
 import ch.njol.skript.util.Time;
 import ch.njol.skript.util.Timeperiod;
 import ch.njol.skript.util.Timespan;
 import ch.njol.util.Kleenean;
 import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.World;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Time")
-@Description("The <a href='classes.html#time'>time</a> of a world.")
-@Examples({"time in world is between 18:00 and 6:00:",
-		"	broadcast \"It's night-time, watch out for monsters!\""})
+@Description({
+	"The <a href='classes.html#time'>time</a> of a world.",
+	"Use the \"minecraft <a href='classes.html#timespan'>timespan</a>\" syntax to change the time according " +
+	"to Minecraft's time intervals.",
+	"Since Minecraft uses discrete intervals for time (ticks), " +
+	"changing the time by real-world minutes or real-world seconds only changes it approximately.",
+	"Removing an amount of time from a world's time will move the clock forward a day."
+})
+@Examples({
+	"set time of world \"world\" to 2:00",
+	"add 2 minecraft hours to time of world \"world\"",
+	"add 54 real seconds to time of world \"world\" # approximately 1 minecraft hour"
+})
 @Since("1.0")
 public class ExprTime extends PropertyExpression<World, Time> {
+
+	// 18000 is the offset to allow for using "add 2:00" without going to a new day
+	// and causing unexpected behaviour
+	private static final int TIME_TO_TIMESPAN_OFFSET = 18000;
+
 	static {
-		Skript.registerExpression(ExprTime.class, Time.class, ExpressionType.PROPERTY, "[the] time[s] [([with]in|of) %worlds%]", "%worlds%'[s] time[s]");
+		Skript.registerExpression(ExprTime.class, Time.class, ExpressionType.PROPERTY,
+			"[the] time[s] [([with]in|of) %worlds%]", "%worlds%'[s] time[s]");
```

Comment:
We could see about using `registerDefault` method instead of the static skript one.

Reply:
i'm for it, but this would be a breaking change. not sure, maybe someone from the team can weigh in?

Reply:
ohh didn't realize `in` and `within` were options, I'd still prefer the change but if we make it might not be able to do dev/patch anymore